src/mono/System.Private.CoreLib/src/System/Reflection/Emit/RuntimeILGenerator.Mono.cs

```diff
@@ -619,10 +619,11 @@ public override void Emit(OpCode opcode, Label[] labels)
         public override void Emit(OpCode opcode, LocalBuilder local)
         {
             ArgumentNullException.ThrowIfNull(local);
-            if (local.ilgen != this)
+            RuntimeLocalBuilder localBuilder = (RuntimeLocalBuilder)local;
+            if (localBuilder.ilgen != this)
```

Comment:
```suggestion
            if (local is not RuntimeLocalBuilder localBuilder || localBuilder.ilgen != this)
```
This will avoid invalid cast exception when somebody passes in LocalBuilder that is not one of ours.

Reply:
(Dtto for other places that cast to RuntimeLocalBuilder.)